src/services/expenses.ts

```diff
@@ -0,0 +1,113 @@
+'use strict'
+
+import { ExpensesStorage } from '../storages'
+import { ServiceRegistry } from './serviceRegistry'
+import { isValidDateString } from '../shared/validation'
+import { ensure } from '../utils'
+import type {
+  RawExpensesFilter,
+  ExpensesFilter,
+  ExpensesInput
+} from '../types'
+
+export default class ExpensesService {
+  private constructor (
+    private storage: ExpensesStorage,
+    private serviceRegistry: ServiceRegistry
```

Comment:
Ideally, a service shouldn't know about the existence of a registry.

Reply:
hm, why? The idea is to use it to get another service's methods. How can we do it w/o having a serviceRegistry as one of the service's properties?

Or do you mean a particular service shouldn't know (in case we have some abstract BaseService class just pass `serviceRegistry` as one of `super` properties, etc?)